core/server/update-check.js

```diff
@@ -38,6 +38,24 @@ var crypto   = require('crypto'),
     checkEndpoint = 'updates.ghost.org',
     currentVersion = config.ghostVersion;
 
+/**
+ * Attempts to parse string to JSON.
+ * @return {Object} is successful. Otherwise, undefined.
+ */
+function safeParseJson(json) {
```

Comment:
If we source it out we can extend the function anytime (like you suggested returning the error in case it throws an exception) and we can use it everywhere - we have a couple of occurrences in Ghost. This is not important at all, so you can choose keeping as it is, sourcing it out or monkey patching `JSON` object. (`JSON.safeParse = function(obj) {...}` in `core/server/overrides`).